src/app/Fake.DotNet.Fsi/Fsi.fs

```diff
@@ -0,0 +1,373 @@
+    /// Define a conditional compilation symbols
+    Define: string
+    /// Ignore ML compatibility warnings
+    MLCompatibility: bool 
+
+(* - MISCELLANEOUS - *)
+    /// Suppress compiler copyright message
+    NoLogo: bool
+    ///  Display the commandline flags and their usage
+    Help: bool
+
+(* - ADVANCED - *)
+    /// Specify the codepage used to read source files
+    Codepage: int option
+    /// Output messages in UTF-8 encoding
+    Utf8Output: bool
+    /// Specify the preferred output language culture name (e.g. es-ES, ja-JP)
+    PreferredUiLang: string
+    /// Output messages with fully qualified paths
+    FullPaths: bool
+    /// Specify a directory for the include path which is used to resolve source files and assemblies (Short form: -I)
+    Lib: string list
+    /// Resolve assembly references using directory-based rules rather than MSBuild resolution
+    SimpleResolution: bool
+    /// Specify target framework profile of this assembly. Valid values are mscorlib, netcore or netstandard. Default - mscorlib
+    TargetProfile: Profile
+    /// Do not reference the default CLI assemblies by default
+    NoFramework: bool
+    /// Exit fsi after loading the files or running the .fsx script given on the command line
+    Exec: bool
+    /// Execute interactions on a Windows Forms event loop (on by default)
+    GUI: bool option
+    /// Suppress fsi writing to stdout
+    Quiet: bool
+    /// Support TAB completion in console (on by default)
+    ReadLine: bool option
+    /// Emit debug information in quotations
+    QuotationsDebug: bool option
+    /// Prevents references from being locked by the F# Interactive process
+    ShadowCopyReferences: bool option
+}
+with 
+    static member ToArgsList p = 
+
+        let stringEmptyMap f s = 
+            if String.isNullOrWhiteSpace s then "" else f s
+
+        /// format a standalone compiler arg: "--%s"
+        let arg s b = if b then sprintf "--%s" s else ""
+        /// format a compiler arg with a parameter: "--%s:%s"
+        let argp s p = stringEmptyMap (sprintf "--%s:%s" s) p            
+        /// format a short form compiler arg with a parameter: "-%s:%s"
+        let sargp s p = stringEmptyMap (sprintf "-%s:%s" s) p  // for short forms          
+        /// helper function to convert a bool to a "+" or "-"
+        let inline chk b = if b then "+" else "-"
+        /// format a compiler arg that ends with "+" or "-": "--%s%s"
+        let togl s b = if Option.isNone b then "" else sprintf "--%s%s" s (chk b.Value)
+        /// format a list of compiler args with string parameters "--%s:\"%s\""   
+        let argls s (ls:string list) = stringEmptyMap (sprintf "--%s:\"%s\"" s) (String.concat ";" ls)
+        /// format a compiler arg that ends with "+" or "-" with string parameters  "--%s%s:\"%s\""
+        let inline toglls s b (ls:'a list) = 
+            stringEmptyMap (sprintf "--%s%s:\"%s\"" s  (chk b)) (String.concat ";" (List.map string ls))
```

Comment:
I guess this doesn't work, have you tried that?
(because `args.ToWindowsCommandLine` is doing the escaping FSI will get a literal quote character).
I think we can just remove the quotes from the sprintf.

I guess the same is for `argls`

Reply:
Ah I guess I didn't test that one. Ill give it a look

Reply:
It looks like it actually works either way, literal quotes in the argument or not -- but I removed them anyways

Reply:
I'd have expected the quotes to be "quoted" again later, strange